sklearn/tree/_tree.pyx

```diff
@@ -438,7 +438,7 @@ cdef class BestFirstTreeBuilder(TreeBuilder):
         n_node_samples = end - start
         is_leaf = ((depth > self.max_depth) or
                    (n_node_samples < self.min_samples_split) or
-                   (n_node_samples < 2 * self.min_samples_leaf) or
+                   (n_node_samples < self.min_samples_leaf) or
```

Comment:
Oh geez that's a typo, sorry I had the logic backwards in my head. Let me think about this a bit more...